.editorconfig

```diff
@@ -14,6 +14,87 @@ indent_size = 4
 [project.json]
 indent_size = 2
 
+# C# files
+[*.cs]
+# brace begins on a new line
+csharp_new_line_before_open_brace = all
+
+# block indentation
+csharp_indent_block_contents = true
```

Comment:
nit: To match the defaults, we should add:
```
csharp_indent_braces = false
csharp_indent_case_contents = true
csharp_indent_switch_labels = true
csharp_indent_labels = no_change
```

I think for `csharp_indent_labels`, the convention is actually `flush_left`